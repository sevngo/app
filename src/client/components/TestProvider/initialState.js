const initialState = {
  auth: {
    _id: '6008432b6ed92c2ac837ec6c',
    username: 'asdasd',
    email: 'asdasd@asdasd.com',
    gender: 'female',
    birthDate: '1991-01-20T00:00:00.000Z',
    address: {
      name: '5 All√©e Claude Chastillon, 93290 Tremblay-en-France, France',
      type: 'Point',
      coordinates: [2.5654428, 48.955157299999996],
    },
    notifications: [
      {
        user: {
          _id: '5caf003feaca23291c0cec34',
          username: 'qweqwe',
        },
        createdAt: '2020-01-23T14:10:55.857Z',
        _id: '5e29a96f7e701b58a77dcdbb',
        messageId: 'friendLogged',
      },
      {
        user: {
          _id: '5caf003feaca23291c0cec36',
          username: 'asdasd',
        },
        createdAt: '2020-01-23T14:11:55.857Z',
        _id: '5e29a96f7e701b58a77dcdba',
        messageId: 'friendLogged',
      },
    ],
    token:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2MDA4NDMyYjZlZDkyYzJhYzgzN2VjNmMiLCJpYXQiOjE2MTEyNTc1MjQsImV4cCI6MTYxMTM0MzkyNH0.suG2IyrXwsI3bOcK1jG7XpasHfwQM1Fip3YrUzX01Pc',
  },
  users: {
    filter: {
      gender: 'male',
      maxDistance: 20000,
      ageRange: [18, 50],
      sortBy: 'distance:asc',
      limit: 10,
      skip: 0,
    },
    data: [
      {
        _id: '60084a46203c4e342b13114c',
        username: 'Shanie21',
        birthDate: '1998-03-03T13:43:52.868Z',
        gender: 'male',
        address: {
          name: '247 Bert Village',
        },
        distance: 2640.691228895585,
      },
      {
        _id: '60084a46203c4e342b13113c',
        username: 'Brianne82',
        birthDate: '1990-05-11T19:00:30.586Z',
        gender: 'male',
        address: {
          name: '4860 Berenice Lodge',
        },
        distance: 2993.4424127582033,
      },
      {
        _id: '60084a46203c4e342b131140',
        username: 'Jon3',
        birthDate: '1973-05-08T21:14:45.718Z',
        gender: 'male',
        address: {
          name: '259 Fabiola Station',
        },
        distance: 6966.4502984182745,
      },
      {
        _id: '60084a46203c4e342b131139',
        username: 'Joannie_Corkery92',
        birthDate: '1998-09-22T09:23:53.485Z',
        gender: 'male',
        address: {
          name: '2431 Marisol Park',
        },
        distance: 9156.496213903085,
      },
      {
        _id: '60084a46203c4e342b13113a',
        username: 'Lilly.MacGyver',
        birthDate: '2002-05-18T07:47:56.337Z',
        gender: 'male',
        address: {
          name: '27384 Carroll Rue',
        },
        distance: 10229.485636305038,
      },
      {
        _id: '60084a46203c4e342b131143',
        username: 'Lacy_Harber23',
        birthDate: '2001-11-18T21:39:20.224Z',
        gender: 'male',
        address: {
          name: '1810 Eliseo Key',
        },
        distance: 11798.426998259116,
      },
      {
        _id: '60084a46203c4e342b131147',
        username: 'Sophie.Gorczany',
        birthDate: '1992-08-23T18:55:03.359Z',
        gender: 'male',
        address: {
          name: '149 Koch Vista',
        },
        distance: 13149.980680446492,
      },
      {
        _id: '60084a46203c4e342b131142',
        username: 'Joelle.Weissnat',
        birthDate: '1985-02-24T06:13:19.138Z',
        gender: 'male',
        address: {
          name: '74795 Howe Inlet',
        },
        distance: 16796.075505426317,
      },
    ],
    total: 8,
  },
  user: {
    _id: '60084a46203c4e342b13114c',
    username: 'Shanie21',
    birthDate: '1998-03-03T13:43:52.868Z',
    gender: 'male',
    address: {
      name: '247 Bert Village',
    },
  },
  loading: true,
  snackbar: {
    variant: 'success',
  },
};

export default initialState;
