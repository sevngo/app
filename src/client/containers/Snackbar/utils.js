import { amber, green } from '@mui/material/colors';
import CheckCircleIcon from '@mui/icons-material/CheckCircle';
import ErrorIcon from '@mui/icons-material/Error';
import InfoIcon from '@mui/icons-material/Info';
import WarningIcon from '@mui/icons-material/Warning';

export const SUCCESS = 'SUCCESS';
export const WARNING = 'WARNING';
export const ERROR = 'ERROR';
export const INFO = 'INFO';

export const variantIcon = {
  [SUCCESS]: CheckCircleIcon,
  [WARNING]: WarningIcon,
  [ERROR]: ErrorIcon,
  [INFO]: InfoIcon,
};

export const variantColor = (variant) => {
  switch (variant) {
    case SUCCESS:
      return { backgroundColor: green[600] };
    case ERROR:
      return { backgroundColor: (theme) => theme.palette.error.dark };
    case WARNING:
      return { backgroundColor: amber[700] };
    default:
      return { backgroundColor: (theme) => theme.palette.primary.main };
  }
};
