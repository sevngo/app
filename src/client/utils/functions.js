import { reduce, filter, isEmpty } from 'ramda';

export const getAge = (dateString) => {
  const today = new Date();
  const birthDate = new Date(dateString);
  let age = today.getFullYear() - birthDate.getFullYear();
  const m = today.getMonth() - birthDate.getMonth();
  if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) age--;
  return age;
};

export const getIds = reduce((acc, object) => [...acc, object._id], []);

export const compact = filter((value) => value && !isEmpty(value));
