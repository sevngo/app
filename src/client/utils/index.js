export * from './validation';
export * from './helpers';
export * from './intl';
export * from './routes';
export * from './theme';
