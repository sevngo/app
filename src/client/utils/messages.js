import { defineMessages } from 'react-intl';

export const formMessages = defineMessages({
  required: {
    id: 'components.input.required',
    defaultMessage: 'Required',
  },
  invalid: {
    id: 'components.input.invalid',
    defaultMessage: 'Invalid',
  },
  minLength: {
    id: 'components.input.tooShort',
    defaultMessage: 'Too short',
  },
  maxLength: {
    id: 'components.input.tooLong',
    defaultMessage: 'Too long',
  },
  tooOld: {
    id: 'components.input.tooOld',
    defaultMessage: 'Too old',
  },
  tooYoung: {
    id: 'components.input.tooYoung',
    defaultMessage: 'Too young',
  },
});
