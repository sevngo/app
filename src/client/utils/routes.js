import { lazy } from 'react';
import { find, prop } from 'ramda';

const Users = lazy(() => import('../containers/Users'));
const User = lazy(() => import('../containers/User'));
const Reset = lazy(() => import('../containers/Reset'));
const Verify = lazy(() => import('../containers/Verify'));
const Login = lazy(() => import('../containers/Login'));
const Register = lazy(() => import('../containers/Register'));

export const usersPath = '/';
export const userPath = (id) => `/user/${id}`;
export const resetPath = (token) => `/reset/${token}`;
export const verifyPath = (token) => `/verify/${token}`;
export const loginPath = '/login';
export const registerPath = '/register';

export const routes = [
  {
    key: usersPath,
    path: usersPath,
    exact: true,
    Component: Users,
    isPrivate: true,
    isDefault: true,
  },
  {
    key: userPath(':id'),
    path: userPath(':id'),
    exact: true,
    Component: User,
    isPrivate: true,
  },
  {
    key: resetPath(':token'),
    path: resetPath(':token'),
    exact: true,
    Component: Reset,
  },
  {
    key: verifyPath(':token'),
    path: verifyPath(':token'),
    exact: true,
    Component: Verify,
  },
  {
    key: loginPath,
    path: loginPath,
    exact: true,
    Component: Login,
    isUnPrivate: true,
  },
  {
    key: registerPath,
    path: registerPath,
    exact: true,
    Component: Register,
    isUnPrivate: true,
  },
];

export const defaultRoute = find(prop('isDefault'))(routes);
