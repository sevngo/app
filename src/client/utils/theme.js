import { createMuiTheme } from '@material-ui/core';

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#4a148c',
    },
    secondary: {
      main: '#e1bee7',
    },
  },
});
