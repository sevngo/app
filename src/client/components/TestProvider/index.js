import { MuiThemeProvider } from '@material-ui/core';
import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter, Route } from 'react-router-dom';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import reducer from '../../reducers';
import { theme } from '../../utils/theme';

jest.mock('../../store');

const TestProvider = ({
  children,
  initialState = {},
  router: { initialEntries, path } = {},
}) => {
  const store = createStore(reducer, initialState, applyMiddleware(thunk));
  return (
    <Provider store={store}>
      <IntlProvider locale="en" messages={{}}>
        <MemoryRouter initialEntries={initialEntries}>
          <Route path={path}>
            <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
          </Route>
        </MemoryRouter>
      </IntlProvider>
    </Provider>
  );
};

export default TestProvider;
