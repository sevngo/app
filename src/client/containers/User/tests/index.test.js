import { render } from '@testing-library/react';
import React from 'react';
import TestProvider from '../../../components/TestProvider';
import { userPath } from '../../../utils';
import Component from '../index';

global.google = {
  maps: {
    places: {
      Autocomplete: function () {
        return {
          setTypes: jest.fn(),
          setFields: jest.fn(),
          addListener: jest.fn(),
        };
      },
    },
  },
};

describe('User', () => {
  it('should render OtherUser', async () => {
    render(
      <TestProvider
        initialState={{ auth: { _id: '60084a46203c4e342b14114c' } }}
      >
        <Component />
      </TestProvider>
    );
  });
  it('should render MyUser', async () => {
    const _id = '60084a46203c4e342b14114c';
    const initialEntries = [`/user/${_id}`];
    const path = userPath(':id');
    render(
      <TestProvider
        initialState={{ auth: { _id } }}
        router={{ initialEntries, path }}
      >
        <Component />
      </TestProvider>
    );
  });
});
