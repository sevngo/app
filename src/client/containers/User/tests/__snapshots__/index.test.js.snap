// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-11"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root makeStyles-mw30-6 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root-143"
          >
            <input
              accept="image/png, image/jpeg"
              class="makeStyles-hide-7"
              type="file"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Delete
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root makeStyles-ml1-2"
                >
                  delete
                </span>
              </span>
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-ml1-2"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Upload
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root makeStyles-ml1-2"
                >
                  cloud_upload
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <img
            alt="profile"
            class="makeStyles-img-9"
            src="emptyImage.png"
          />
          <div
            class="MuiPaper-root MuiMobileStepper-root MuiMobileStepper-positionStatic MuiPaper-elevation0"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root"
                >
                  keyboard_arrow_left
                </span>
                Back
              </span>
            </button>
            <div
              class="MuiMobileStepper-dots"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Next
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root"
                >
                  keyboard_arrow_right
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-mw30-6 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiPaper-root makeStyles-p3-1 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <form>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
                data-shrink="true"
              >
                Username
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root makeStyles-icon-214"
                >
                  account_circle
                </span>
                <input
                  aria-invalid="false"
                  autocomplete="username"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  name="username"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-273 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-275 PrivateNotchedOutline-legendNotched-276"
                  >
                    <span>
                      Username
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
                data-shrink="true"
              >
                New password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root makeStyles-icon-214"
                >
                  vpn_key
                </span>
                <input
                  aria-invalid="false"
                  autocomplete="password"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  name="newPassword"
                  type="password"
                  value=""
                />
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <span
                      aria-hidden="true"
                      class="material-icons MuiIcon-root"
                    >
                      visibility_off
                    </span>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-273 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-275 PrivateNotchedOutline-legendNotched-276"
                  >
                    <span>
                      New password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
                data-shrink="true"
              >
                Email
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root makeStyles-icon-214"
                >
                  alternate_email
                </span>
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  name="email"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-273 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-275 PrivateNotchedOutline-legendNotched-276"
                  >
                    <span>
                      Email
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
                data-shrink="true"
              >
                Birth date
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <span
                  aria-hidden="true"
                  class="material-icons MuiIcon-root makeStyles-icon-214"
                >
                  date_range
                </span>
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  name="birthDate"
                  type="date"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-273 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-275 PrivateNotchedOutline-legendNotched-276"
                  >
                    <span>
                      Birth date
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiBox-root MuiBox-root-286"
            >
              <div
                class="MuiFormControl-root"
              >
                <label
                  class="MuiFormLabel-root"
                >
                  Gender
                </label>
                <div
                  class="MuiFormGroup-root MuiFormGroup-row"
                  role="radiogroup"
                >
                  <label
                    class="MuiFormControlLabel-root"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-300 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-303"
                          name="gender"
                          type="radio"
                          value="male"
                        />
                        <div
                          class="PrivateRadioButtonIcon-root-304"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                            />
                          </svg>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-305"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                            />
                          </svg>
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Male
                    </span>
                  </label>
                  <label
                    class="MuiFormControlLabel-root"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-300 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-303"
                          name="gender"
                          type="radio"
                          value="female"
                        />
                        <div
                          class="PrivateRadioButtonIcon-root-304"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                            />
                          </svg>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-305"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                            />
                          </svg>
                        </div>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Female
                    </span>
                  </label>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
                data-shrink="false"
                for="address"
                id="address-label"
              >
                Address
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  id="address"
                  name="address.name"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-273 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-275"
                  >
                    <span>
                      Address
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-mt1-208 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled MuiButton-fullWidth Mui-disabled"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Submit
              </span>
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge Mui-disabled MuiButton-fullWidth Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Cancel
              </span>
            </button>
          </form>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-mw30-6 MuiGrid-item MuiGrid-grid-xs-12"
      />
    </div>
  </div>
</div>
`;
